templates/step/viewimg.html.ep

```diff
@@ -1,23 +1,43 @@
 <div class="aligncenter" data-image="<%= url_for('test_img', filename => $screenshot) %>" id="step_view">
-    Candidate needle:
-    <select id="needlediff_selector">
-        <option data-areas="[]" data-matches="[]">-None-</option>
-        % for my $needle (@$needles) {
-            % my $title = $needle->{min_similarity} . "%: " . $needle->{name};
-            <option data-image="<%= $needle->{image} %>"
-                    data-areas="<%= JSON::to_json($needle->{areas})%>"
-                    % if ($needle->{selected}) {
-                    selected="selected"
-                    % }
-                    data-matches="<%= JSON::to_json($needle->{matches}) %>">
-                <%= $title %></option>
-        % }
-    </select>
+    <span id="candidate_needles">
+        Candidate needle:
+        <select id="needlediff_selector">
+            <option data-areas="[]" data-matches="[]">-None-</option>
+            % for my $needle (@$needles) {
+                % my $title = $needle->{min_similarity} . "%: " . $needle->{name};
+                <option data-image="<%= $needle->{image} %>"
+                        data-areas="<%= JSON::to_json($needle->{areas})%>"
+                        % if ($needle->{selected}) {
+                        selected="selected"
+                        % }
+                        data-matches="<%= JSON::to_json($needle->{matches}) %>">
+                    <%= $title %></option>
+            % }
+        </select>
 
-    % if (is_operator) {
-        %= link_to 'Create new needle' => url_for('edit_step')
-    % }
+        % if (is_operator) {
+            %= link_to 'Create new needle' => url_for('edit_step')
+        % }
+    </span>
+    <span style="float;right;" id="bug_report">
+        % my $config = $self->app->config->{external_reporting};
+        % if ($config) {
+            Report
+            % if ($config->{product_tracker} && $config->{product_tracker} eq 'bugzilla') {
+                % my $product_bug_subject = $config->{product_subject} // '[Build ' . $job->BUILD . '] openQA test fails';
+                % my $product_bug_comment = $config->{product_description} // "DESCRIPTION GOES HERE";
+                % my $mappings = $self->app->config->{'product_mapping_' . $config->{product_tracker}};
+                % my $product = $mappings->{$job->group_id};
+                %= link_to 'product bug' => url_for($config->{product_url_new})->query(product => $product, short_desc => '[Build ' . $job->BUILD . '] openQA test fails', bug_file_loc => url_for('test_img')->to_abs, comment=> $product_bug_comment)
+                /
+            % }
+            % if ($config->{test_tracker} && $config->{test_tracker} eq 'redmine') {
```

Comment:
you should not add that much code to views. There are helpers for that - which also makes it possible to create unit tests for that functions.